src/Rules.Framework/Providers/InMemory/RuleFactory.cs

```diff
@@ -18,13 +18,13 @@ internal sealed class RuleFactory<TContentType, TConditionType> : IRuleFactory<T
                 throw new ArgumentNullException(nameof(ruleDataModel));
             }
 
-            var contentContainer = new ContentContainer<TContentType>(ruleDataModel.ContentType, (_) => ruleDataModel.Content);
-            var ruleBuilderResult = RuleBuilder.NewRule<TContentType, TConditionType>()
+            var ruleBuilderResult = Rules.Framework.RuleBuilder
+                .NewRule<TContentType, TConditionType>()
                 .WithName(ruleDataModel.Name)
                 .WithActive(ruleDataModel.Active)
                 .WithDatesInterval(ruleDataModel.DateBegin, ruleDataModel.DateEnd)
                 .WithCondition(cnb => ruleDataModel.RootCondition is { } ? ConvertConditionNode(cnb, ruleDataModel.RootCondition) : null)
-                .WithContentContainer(contentContainer)
+                .WithContent(ruleDataModel.ContentType, ruleDataModel.Content)
```

Comment:
Try casting the content to `object` to resolve your build issue 🙂

Reply:
It worked! thanks 🙂